src/foam/approval/ApprovableAwareDAO.js

```diff
@@ -0,0 +1,450 @@
+foam.CLASS({
+  package: 'foam.approval',
+  name: 'ApprovableAwareDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.core.X',
+    'foam.core.PropertyInfo',
+    'java.util.Iterator',
+    'foam.dao.DAO',
+    'java.util.Map',
+    'java.util.List',
+    'java.util.Set',
+    'foam.mlang.MLang',
+    'foam.mlang.MLang.*',
+    'foam.core.FObject',
+    'foam.dao.ArraySink',
+    'java.util.ArrayList',
+    'foam.util.SafetyUtil',
+    'foam.nanos.auth.User',
+    'foam.core.Detachable',
+    'foam.dao.AbstractSink',
+    'foam.nanos.logger.Logger',
+    'foam.lib.PropertyPredicate',
+    'foam.nanos.ruler.Operations',
+    'foam.nanos.auth.LifecycleState',
+    'foam.nanos.auth.LifecycleAware',
+    'foam.mlang.predicate.Predicate',
+    'foam.approval.ApprovalStatus',
+    'foam.approval.ApprovalRequest',
+    'foam.comics.v2.userfeedback.UserFeedbackAware',
+    'foam.comics.v2.userfeedback.UserFeedback',
+    'foam.comics.v2.userfeedback.UserFeedbackStatus',
+    'foam.comics.v2.userfeedback.UserFeedbackException',
+    'foam.nanos.crunch.UCJQueryService',
+    'foam.approval.Approvable',
+    'foam.approval.ApprovableAware',
+    'foam.approval.RoleApprovalRequest'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'daoKey'
+    },
+    {
+      class: 'Class',
+      name: 'of'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'sendSingleRequest',
+      type: 'void',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'req', type: 'RoleApprovalRequest' },
+        { name: 'userId', type: 'long' }
+      ],
+      javaCode: `
+        RoleApprovalRequest request = (RoleApprovalRequest) req.fclone();
+        request.clearId();
+        request.setApprover(userId);
+        ((DAO) x.get("approvalRequestDAO")).put_(x, request);
```

Comment:
add approvalRequestDAO to the FOAM services file.

Reply:
Done.